### PR TITLE
Add datepicker coverage specs to 88%

### LINKED ISSUE #247: Coverage: datepicker to 90%
## Parent

GSA/ngx-uswds#237

## What to build

Cover the `datepicker` component (19 source files, no spec) — the largest untested surface in the library. Include date parsing/formatting, calendar navigation, min/max constraints, and keyboard interaction.

Add or extend the spec(s) so the unit(s) are exercised end-to-end: component construction, all @Input/@Output bindings, template rendering for each state, and user interactions (clicks, keyboard, focus). Cover branch paths, not just happy-path lines. Land a measurable coverage gain that keeps the library at or above the current coverage floor.

**Coverage-gate policy (changed in #259):** Do **not** edit the coverage gate or its thresholds in this PR. The floors now live in `coverage-floor.json`, and feature/test PRs must not touch them — that shared edit is what made the parallel coverage PRs conflict. Your PR only needs to keep coverage **at or above** the current floor. Raising the ratchet is a separate, deliberate step (`npm run coverage:bump`) landed on its own; a maintainer will bump the floor after slices merge.

## Acceptance criteria

- [ ] Spec(s) added/extended for the target unit(s)
- [ ] Statements, branches, functions, and lines meaningfully increased for this area
- [ ] Coverage stays at or above the current floor in `coverage-floor.json` (do **not** edit the gate/floors in this PR — see #259)
- [ ] All tests pass in CI

## Blocked by

- GSA/ngx-uswds#238 (Vitest migration + gate)

### PR DESCRIPTION
**Pull request by fpigeonjr**
## Description

Adds 11 new spec files comprehensively covering the `datepicker` module — previously the largest untested surface in the library (19 source files, zero specs).

**By area:**

- **`dateadapter/native-date-adapter.spec.ts`** (51 tests): parse, format, calendar arithmetic (days/months/years), ISO 8601 deserialization, edge cases (leap years, month overflow, invalid dates)
- **`dateadapter/date-adapter.spec.ts`** (23 tests): `compareDate`, `sameDate`, `clampDate`, `getValidDateOrNull`, `deserialize`, `setLocale`/`localeChanges`
- **`date-selection-model.spec.ts`** (31 tests): `DateRange`, `UsaSingleDateSelectionModel`, `UsaRangeDateSelectionModel` — add/isValid/isComplete/clone/selectionChanged/ngOnDestroy; factory functions
- **`date-range-selection-strategy.spec.ts`** (10 tests): `DefaultUsaCalendarRangeStrategy.selectionFinished` and `createPreview`; provider factory
- **`date-picker-errors.spec.ts`** (3 tests): `createMissingDateImplError`
- **`calendar.spec.ts`** (17 tests): `UsaCalendar` view switching, date selection, min/max inputs, relay outputs, `ngOnChanges`, `updateTodaysDate`, `_goToDateInView`
- **`calendar/month-view.spec.ts`** (25 tests): full keyboard navigation (ArrowLeft/Right/Up/Down/Home/End/PageUp/PageDown ± Alt), Space/Enter selection, `dateFilter`, `selected` input
- **`calendar/year-view.spec.ts`** (25 tests): same keyboard suite at year granularity, month selection, min/max disable logic
- **`calendar/multi-year-view.spec.ts`** (28 tests): same keyboard suite at decade granularity, year selection, `getNextYears`/`getPreviousYears`, `_getActiveCell`
- **`date-picker-input.spec.ts`** (17 tests): `UsaDatePickerInput` CVA, disabled state, `getConnectedOverlayOrigin`, Alt+ArrowDown open, `_onInput`/`_onChange`/`_onBlur`, `stateChanges`
- **`date-picker-button.spec.ts`** (6 tests): `UsaDatePickerButton` construction, disabled inheritance, `_toggle` guard (null picker, disabled)

**Coverage improvement:**

| Metric | Floor | Before | After |
|---|---|---|---|
| Statements | 81% | 81% | **88.21%** |
| Branches | 83% | 83% | **88.67%** |
| Functions | 48% | 48% | **72.03%** |
| Lines | 81% | 81% | **88.21%** |

`coverage-floor.json` is **not edited** in this PR per the ratchet policy (#259). A maintainer can run `npm run coverage:bump` to lock in the new floors after this merges.

## Motivation and Context

Closes #247

## Type of Change (Select One and Apply Label)

- [ ] Bug fix (non-breaking change which fixes an issue) → Apply `bugfix` label
- [x] New feature (non-breaking change which adds functionality) → Apply `enhancement` label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) → Apply `breaking` label
- [ ] Documentation / configuration update → Apply `documentation` label

## How to Test

1. `cd` into the worktree and install: `npm ci`
2. Run the test suite with coverage: `npm run test:components`
3. Verify all 548 tests pass (0 failures)
4. Run the coverage gate: `npm run coverage:check`
5. Confirm all four metrics are above their floors

**Expected result:** `✓ Coverage gate passed.` with statements/lines ≥81%, branches ≥83%, functions ≥48%.

## Screenshots (if appropriate)

N/A — test-only changes, no UI modifications.

## Checklist

- [x] Branch name follows convention (e.g. `gh-<number>-<slug>`)
- [x] PR title starts with a verb in the imperative mood
- [x] I have self-reviewed my own code
- [x] `format:check` passes (`npm run format:check`)
- [ ] `lint` passes (`npm run lint`) — lint target not configured in this repo (known issue per AGENTS.md)
- [ ] `build-prod` passes (`npm run build-prod`)
- [x] Tests pass and coverage is reported (`npm run test:components` — 548 passed)
- [ ] If this change requires a documentation update, I have updated it accordingly
- [ ] If there are dependent changes, they have been merged and published in downstream modules

### CODE REVIEW

File: projects/uswds-components/src/lib/datepicker/dateadapter/date-adapter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import { DateAdapter, USA_DATE_LOCALE } from './date-adapter';
+import { NativeDateAdapter } from './native-date-adapter';
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply projects/uswds-components/src/lib/datepicker/dateadapter/date-adapter.spec.ts]: Fixed in b4c706d: removed `USA_DATE_LOCALE` from the import (only `DateAdapter` is used — to name the class in the describe label).

File: projects/uswds-components/src/lib/datepicker/date-selection-model.spec.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+import {
+  DateRange,
+  UsaSingleDateSelectionModel,
+  UsaRangeDateSelectionModel,
+  UsaDateSelectionModel,
+  USA_SINGLE_DATE_SELECTION_MODEL_PROVIDER,
+  USA_RANGE_DATE_SELECTION_MODEL_PROVIDER,
+  USA_SINGLE_DATE_SELECTION_MODEL_FACTORY,
+  USA_RANGE_DATE_SELECTION_MODEL_FACTORY,
+} from './date-selection-model';
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply projects/uswds-components/src/lib/datepicker/date-selection-model.spec.ts]: Fixed in b4c706d: removed unused `TestBed`, `DateAdapter`, `UsaDateSelectionModel`, `USA_SINGLE_DATE_SELECTION_MODEL_PROVIDER`, and `USA_RANGE_DATE_SELECTION_MODEL_PROVIDER` from the imports.

File: projects/uswds-components/src/lib/datepicker/date-picker-input.spec.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import { UsaDatePickerInput } from './date-picker-input';
+import { KeyCode } from '../util/key';
+
+@Component({
+  standalone: false,
+  template: `
+    <usa-date-picker-wrapper>
+      <input
+        #input
+        usaDatePicker
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply projects/uswds-components/src/lib/datepicker/date-picker-input.spec.ts]: Fixed in b4c706d: changed the host template to `[usaDatePicker]="picker"` so the @Input setter receives the component reference and registers it. Also replaced the `if (!picker) return;` guards with `expect(picker).toBeTruthy()` so any future regression fails fast instead of silently passing.

File: projects/uswds-components/src/lib/datepicker/calendar.spec.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    it('switches to year view and emits viewChanged', () => {
+      // calendar starts in month; switching to year should emit
+      const before = host.viewChangedValues.length;
+      calendar.currentView = 'year';
+      fixture.detectChanges();
+      expect(calendar.currentView).toBe('year');
+      // viewChanged is internal to UsaCalendar — it uses its own EventEmitter,
+      // not the host binding.  Verify the view changed.
+      expect(calendar.currentView).toBe('year');
+    });
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply projects/uswds-components/src/lib/datepicker/calendar.spec.ts]: Fixed in b4c706d: wrapped the test in `fakeAsync`/`tick()` to flush the async EventEmitter (`new EventEmitter(true)`), then added assertions that `host.viewChangedValues` grew and that the last emitted value is `'year'`. The test now verifies the output binding fired.

**Review by divyabairavarasu — APPROVED** [review]
